src/simplify_algebra.cpp

```diff
@@ -439,13 +498,48 @@ struct find_mul_add
         auto b_ins = r.instructions["b"];
         auto x_ins = r.instructions["x"];
         assert(x_ins != b_ins);
-
+    
         auto ax_ins = m.insert_instruction(ins, make_op("mul"), a_ins, x_ins);
         auto ab_ins = m.insert_instruction(ins, make_op("mul"), a_ins, b_ins);
         m.replace_instruction(ins, make_op("add"), ax_ins, ab_ins);
+
     }
 };
 
+
+struct find_slice_add_mul
```

Comment:
Add a description of what this matcher is looking for and what it's trying to do.